odoo/models.py

```diff
@@ -2246,11 +2246,17 @@ def _read_group_prepare(self, orderby, aggregated_fields, annotated_groupbys, qu
             for key in ('field', 'groupby')
         }
         for order_part in orderby.split(','):
-            order_split = order_part.split()
+            order_split = order_part.split()  # potentially ["field:agg", "desc"]
             order_field = order_split[0]
+            if '.' in order_field:
+                order_field, is_many2one_id = order_field.split(".")
+            else:
+                is_many2one_id = None
```

Comment:
I would be more specific on two things: `is_many2one_id` is a boolean, and `order_field` should end with `".id"`
```suggestion
            is_many2one_id = order_field.endswith('.id')
            if is_many2one_id:
                order_field = order_field[-3:]
```